e2e/nomostest/config_sync.go

```diff
@@ -666,10 +670,21 @@ func RootSyncObjectV1Alpha1(name, repoURL string, sourceFormat filesystem.Source
 		Repo:   repoURL,
 		Branch: gitproviders.MainBranch,
 		Dir:    gitproviders.DefaultSyncDir,
-		Auth:   "ssh",
-		SecretRef: &v1alpha1.SecretReference{
+	}
+	switch *e2e.GitProvider {
+	case e2e.CSR:
+		// CSR provider requires auth because the repository is private
+		if *e2e.GceNode {
+			rs.Spec.Git.Auth = configsync.AuthGCENode
+		} else {
+			rs.Spec.Git.Auth = configsync.AuthGCPServiceAccount
+			rs.Spec.Git.GCPServiceAccountEmail = gitproviders.CSRReaderEmail()
```

Comment:
>  I think it's a smell how there isn't any gcenode specific setup/assertions in TestGCENode

We do have an assertion for the gcenode tests: `workloadidentity.ValidateDisabled(nt)`. Maybe the `e2e.GceNode` flag name is confusing? All CIs are running on GCE nodes, but gcenode tests require WI to be disabled. What about renaming to something like `--wi=disabled`?

>  If --gcenode --git-provider=csr is provided, I should in theory be able to run any of the generic git-sync tests with that config.

I'm not sure if I agree with this. We just need a single test to validate gcenode + CSR works. We don't need to run the entire test suite to test all features with gcenode+CSR.

> If the idea is that we don't want to run the whole test suite for the CI job, then we have other options to filter which tests are run (either by test name regex or test group).

Yes. I think we can rename `TestHelmGCENode` to `TestGCENodeHelm` and reuse  `-run=TestGCENode*` to filter these gcenode tests.

Reply:
> We do have an assertion for the gcenode tests: workloadidentity.ValidateDisabled(nt). Maybe the e2e.GceNode flag name is confusing? All CIs are running on GCE nodes, but gcenode tests require WI to be disabled. What about renaming to something like --wi=disabled?

I think `workloadidentity.ValidateDisabled(nt)` is perhaps redundant and/or should just be handled by the test scaffolding. The test scaffolding already handles the disabling of workload identity on cluster creation when `--gcenode` is set. I don't think there's any need to rename the parameter.

> I'm not sure if I agree with this. We just need a single test to validate gcenode + CSR works. We don't need to run the entire test suite to test all features with gcenode+CSR.

I'm not suggesting that we necessarily create a CI job that runs the whole test suite. But with these abstractions in place I don't see a technical reason that I shouldn't be able to run any generic `git-sync` test using gcenode+CSR. For example if I wanted to run some one-off test from my workstation with that permutation.

> Yes. I think we can rename TestHelmGCENode to TestGCENodeHelm and reuse -run=TestGCENode* to filter these gcenode tests.

How about running the `sync-source` test group?

Reply:
> The test scaffolding already handles the disabling of workload identity on cluster creation when --gcenode is set. 

Cool. I wasn't aware of that. Then agreed that we don't need the assertion.

> How about running the sync-source test group?

Then do we run all tests in the `sync-souce` test group with gcenode CSR/AR?

Reply:
Forgot to reply your comment regarding `I think workloadidentity.ValidateDisabled(nt) is perhaps redundant and/or should just be handled by the test scaffolding.`.

I think we still need the validation because we may run the test with a pre-provisioned GKE cluster, which may have WI enabled.